# Google Play and App Store Profitable Apps

In this project I will analyze app profiles for both the Google Play and App Store markets to do be able determine which types of apps attract the most users. 

Working as a data analyst in a company which only builds apps that are free to download and install, our source of revenue comes from user interaction with in-app ads. By helping app developers identify the apps with the most users we will be able to maximize our revenue; the more users and app has the more interactions with in-apps ads. Our goal to aid app developers in their decisions of which apps to develop in order to attract the most users. 

# Opening & Exploring Data

As of September 2018, there were approximately 2 million apps in the App Store, and 2.1 million apps in Google Play.

Analyzing data for over 4 million apps would involve a consumption of a large amount of resources, for this reason we are going to analyze data for about 10,000 apps each from the App Store and Google Play. 

- Data for the App Store can be downloaded at this [Link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)

- Data for Google Play can be downloaded at this [Link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)


Let's begin by opening and reading in both datasets. 

In [24]:
##  Open and read in both datasets  ##

from csv import reader

# Google Play Data #

opened_file = open('googleplaystore.csv', encoding='utf8')
read_file = reader(opened_file)
google = list(read_file)
gplay_header = google[0]
gplay_data = google[1:]


# App Store Data #

opened_file = open('AppleStore.csv', encoding='utf8')
read_file = reader(opened_file)
app_store = list(read_file)
app_store_header = app_store[0]
app_store_data = app_store[1:]

To make data exploration easier, we will define a function `explore_data()` which that we can use to easily explore the rows of the dataset, as well as give an option to print the number of rows and columns within each. The `explore_data()` function takes as parameters the dataset you wish to explore, the start and end index of the rows you would like to print, and the option (True/False) to print the number of rows and columns within the dataset.

In [25]:
# Define function explore_data

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:   # if you specify True in parameters will print number of rows and columns, default is False
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        print('\n')

We can begin the task of exploring the dataset and figuring out which columns within the App Store and Google Play datasets will help us in defining what types apps attract most users. 

In case you would like to review the full documenation of the dataset to better understand the information found within each column:

- Apple Store documentation can be found at this [Link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)
- Google Play documentation can be found at this [Link](https://www.kaggle.com/lava18/google-play-store-apps)

We will first explore the Google Play store colunms and first five rows.

In [29]:
# Display Google Play data columns
print(gplay_header)
print('\n')

# Explore first five rows of Google Play data
# I have added the True paremeter so it prints the number of rows and columns in the dataset
explore_data(gplay_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

We can see that the Google Play data is made up of 10,841 rows and 13 columns. Some of the columsn that will be useful for our analysis are: 

- `'App'`, `'Category'`, `'Rating'`, `'Reviews'`, `'Installs'`, `'Type'`, `'Price'`, and `'Genre'`

In case you need more clarification on what each of these rows contains you can find the documentation at this [Link](https://www.kaggle.com/lava18/google-play-store-apps).

We will now explore the App Store columns and first five rows. 

In [30]:
# Display App Store data columns
print(app_store_header)
print('\n')

# Explore first five rows of App Store data
# I have added the True paremeter so it prints the number of rows and columns in the dataset
explore_data(app_store_data, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16




We can see that the App Store dataset is made up of 7,197 rows and 16 columns. Some of the columns that will help with our analysis are:

- `'track_name'`, `'currency'`, `'price'`, `'rating_count_tot'`, `'rating_count-ver'`, and `'prime_genre'`

Again, if you would like more specifics on what these rows contain you can find the documentation at this [Link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

# Cleaning the Datasets

In the previous steps we explored the first five rows and columns of each dataset. Before continuing with our analysis though, we need to make sure that we have clean data for accurate analysis. This means:

- Finding and removing innacurate data
- Finding and removing duplicate data

For our project we only want to analyze data from apps that are free and marketed towards English speaking users. For this reason, we will need to remove:

- Apps that are not free
- Apps that are in other languages

Once we have performed the data cleaning process, we will have a dataset we can use for analyes. 

## Google Play Innacurate Data Clean Up

We will begin by finding and removing innacurate data from Google Play store. 

If you look at the [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion) of the Google Play store data, we can see that [one discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) describes an error for entry 10472. There is a missing entry in the `'Category'` column, and this has caused the rest of entrys for this row to shift one column back. 

We will begin by looking at the specific row in question in more detail. 

In [27]:
# Detail of row 10472 of Google Play dataset
print(gplay_data[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


Upon inspection of row 10472, we can see that there are only 12 variables withing this row. If you look back at the number of rows above, we see that there are 13 rows, hence one row `'Category'` is missing which renders this row innacurate.

We will delete this row to in order to leave us with a clean Google Play dataset. Be careful to not run the next cell more than once. After you delete row 10472, a new row takes the same index; everytime you run the cell you would be deleting that new row.

In [28]:
# Delete row 10472
del(gplay_data[10472])

# Check that the one row has been deleted. Original dataset is 10841 rows, after deletion should be 10840
print(len(gplay_data))

10840


## App Store Innacurate Data Clean Up

As we did with the Google Play data, we will check the [discussion boards](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion) to find out what we might need to clean up in this dataset. 

After inspection of the [discussion boards](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/discussion), we can see that there is no innacurate data. It does mention duplicate data but we will deal with this issue next. 

## Google Play Duplicate Data Clean Up

In the previous steps we cleaned up the innacurate data, but if you look into the [discussions boards](https://www.kaggle.com/lava18/google-play-store-apps/discussion) there are various mentions of duplicate data. Duplicate data would be the same app being found in many rows. 

Let's begin by checking the Google Play data for duplicate apps. 

In [33]:
# Begin by creating empty lists that will detail duplicate/unique apps

duplicate_apps = []
unique_apps = []

# We will check for unique & duplicate apps with a for loop

for app in gplay_data:
    name = app[0]  # this is the index where the app name is found
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
# Print the number of duplicate and some exmaples

print('Number of Duplicate Apps:', len(duplicate_apps))
print('\n')
print('Examples of Duplicate Apps:', duplicate_apps[:15])

Number of Duplicate Apps: 1181


Examples of Duplicate Apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


As we can see, there a total of 1,181 duplicate entries. Let's explore one of the apps with duplicates to figure out why there are duplicates. 

In [37]:
for app in gplay_data:
    name = app[0]
    if name == 'Google Ads':
        print(app)

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']
['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


The difference is encountered in the fourth column `'Reviews'`, the number of reviews increases from 29,313 to 29,331. The duplicates might happen as more up to date is received and updated into the dataset. What we shall do here keep the row with highest number of reviews and delete the rest, so that we can analyze the most up to date data. 

After removing the 1,181 duplicates found we should end up with a total of 9,659 rows, as you can see below.

In [38]:
print('Lenght After Removal:', len(gplay_data) - 1181)

Lenght After Removal: 9659


To remove the duplicates, we will create a dictionary where each key is a unique app and each value represents the highest number of reviews. Using this dictionary we will create a new data set which will contain only one entry per app. 

Begin by creating dictionary `reviews_max` with app name as keys and max number of reviews as values. 

In [44]:
# Make empty dicitionary to save data
reviews_max = {}

# Loop through Google Play dataset to find apps and max number of reviews
for app in gplay_data:   
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
# Inspect reviews to make sure we have correct amount of entries: 9,659
print(len(reviews_max))   

9659


As you can see the `reviews_max` dictionary is of correct lenght: 9,659. Now we can use the `reviews_max` dictionary to remove the duplicate rows and create a clean dataset `gplay_clean`. In `gplay_clean` you will only find one row per app containing the max number of reviews. 

Start by creating to empty lists, `gplay_clean` and `already_added`. We fill in `gplay_clean` by isolating each app and app name, adding the whole app row to `gplay_clean'` only if it contains the max number of reviews and isn't in `already_added`. 

In [46]:
# Create list to store new clean data
gplay_clean = []

# Create list to store app names that are already in gplay_clean
already_added = []

# Loop through Google Play dataset to create the clean dataset
for app in gplay_data:
    name = app[0]
    n_reviews = float(app[3])
    
    if reviews_max[name] == n_reviews and name not in already_added:
        gplay_clean.append(app)
        already_added.append(name)

Let's now explore our new clean dataset and confirm we have 9,659 rows by using the `explore_data()` function defined in previous steps. 

In [48]:
# Use explore_data() function
explore_data(gplay_clean, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13




As expected, we now have a clean dataset containing 9,659 rows. 

## Removing Non-English Apps

In the previous steps we cleaned up wrong and duplicate data. 

Our company develops apps strictly directed at an English-speaking audience. For this reason, we want to explore the data and remove any non English apps that are present in the datasets. 

What we are tyring to accomplish is to remove any app that has characters not present in the English alphabet (letters, numbers, punctuation marks, symbols). 

Each character we use in a string has a corresponding number associated with it. We can find this number by using the `ord()` fucntion. 

In [54]:
print(ord('a'))
print(ord('b'))
print(ord('A'))
print(ord('B'))

97
98
65
66


The numbers corresponding to the characters we normally see in the English language are in a range from 0 to 127 according to [ASCII](https://en.wikipedia.org/wiki/ASCII) (American Standard Code for Infomation Interchange) system. 

Based on this range we will build a function that detects English characters and those that are not common English characters. If an app name contains character greater than 127 then it probably isn't an English character. 

Our app names are stored as strings, and strings like lists or dictionaries are indexable and iterable, which we can iterate through app names with a for loop.

In [103]:
string_a = 'google'
print(string_a[0])
print(string_a[1])
print(string_a[2])
print('\n')

for character in string_a:
    print(character)

g
o
o


g
o
o
g
l
e


Let's first define a function `eng_string()` that takes in a string and returns False if there are any non English characters and True if no non English characters. 

The function will iterate over each character of the string and check if that character is associated with a number greater than 127. 

We will check if the function works with the following app names:

- 'Instagram'
- '爱奇艺PPS -《欢乐颂2》电视剧热播'
- 'Docs To Go™ Free Office Suite'
- 'Instachat 😜'

In [122]:
# Defining eng_string()

def eng_string(string):
    
    for x in string:
        if ord(x) > 127:
            return False
        
    return True

print(eng_string('Instgarm'))
print(eng_string('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(eng_string('Docs To Go™ Free Office Suite'))
print(eng_string('Instachat 😜'))

True
False
False
False


We can see that the function uncorrectly identifies 'Docs To Go™ Free Office Suite' and 'Instachat 😜' as non English apps, reason being the '™' and 😜 characters. 

For this reason we will redefine the `eng_string()` function to include a parameter where if the app name more than 3 non English then it will return False and label it a non Egnlish app. 

In [123]:
# Defining eng_string()

def eng_string(string):
    non_eng = 0
    
    for x in string:
        if ord(x) > 127:
            non_eng += 1
            
    if non_eng > 3:
        return False
    
    else:
        return True
        
print(eng_string('Instgarm'))
print(eng_string('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(eng_string('Docs To Go™ Free Office Suite'))
print(eng_string('Instachat 😜'))

True
False
True
True


Now we will use the `eng_string()` function to check each of our datasets. If an app name is identified as English we will append it to a new list `english_apps`. 

In [124]:
# Creating English only Google Play dataset

gplay_eng_apps = []

for app in gplay_clean:
    name = app[0]
    
    if eng_string(name) == True:
        gplay_eng_apps.append(app)
        
# Creating English only App Store dataset

app_store_eng_apps = []

for app in app_store_data:
    name = app[1]
    
    if eng_string(name) == True:
        app_store_eng_apps.append(app)        

Let's now explore our datasets with the previously defined `explore_data()` function to see how many apps we have left. 

In [125]:
# Explore Google Play Data
explore_data(gplay_eng_apps, 0, 5, True)

# Explore App Store Data
explore_data(app_store_eng_apps, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', 

We can see that we are now left with 9,614 Google Play apps and 6,183 App Store apps. 

## Removing Non Free Apps

We have so far removed innacurate data, duplicate apps, and non-English apps. 

Our company only develops apps that are free to download and install. The next step in our process will be to remove those apps that are not free. 

In [130]:
# Cleaning non free Google Play apps

gplay_final= []

for app in gplay_eng_apps:
    price = app[7]
    if price == '0':
        gplay_final.append(app)
        
#Cleaning non free App Store apps

app_store_final = []

for app in app_store_eng_apps:
    price = app[4]
    if price == '0.0':
        app_store_final.append(app)

Let's now explore the new clean datasets.

In [132]:
# Explore Google Play Data
explore_data(gplay_final, 0, 5, True)

# Explore App Store Data
explore_data(app_store_final, 0, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 8864
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', 

We are left with a total of 8,864 Google Play apps and 3,222 App Store apps.

# Dataset Analysis - Most Common App Genres

We now have two clean datasets from which we removed innacurate data, duplicate apps, non-English apps, and non free apps. 

Our aim in this analysis is to determine the kinds of apps that have the most users given that our app revenue depends highly on the number of users. 

To minimize the risk and costs associated with developing an app we have a three strep strategy:

- Build minimal Android version and add it to Google Play
- If app has good response, develop it further
- If app is profitable after six months, build iOS version for App Store

Our end goal is to add the app to both markets, for this reason we need to develop profiles for apps that work well in both. 

We will begin by analyzing the most common app genres for each market. 

Let's inspect the the datasets to identify columns that might be of use in analyzing the most common app genres. 

In [134]:
print(gplay_header)
print('\n')
print(app_store_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


For the Google Play data we can use 'Category' and 'Genres' columns to build a frequency table.

For the App Store data we can use the 'prime_genre' column to build a frequency table.

We will build two functions to analyze the frequency tables. 

- One function `freq_table()` to generate a frequency table.
- One function `display_table()` to display percentages in descending order.

The `display_table()` function will take the table generarted by `freq_table()` transform it into a list of tuples and display the entries in descending order. 

In [144]:
# Defining freq_table() function

def freq_table(dataset, index):
    temp_freq = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        
        if value in temp_freq:
            temp_freq[value] += 1
        else:
            temp_freq[value] = 1
            
    table_perc = {}
    
    for key in temp_freq:
        perc = (temp_freq[key] / total) * 100
        table_perc[key] = perc
            
    return table_perc


# Display display_table() function

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', round(entry[0], 2))

## App Store Genres

We start by analyzing the `prime_genre` column in the App Store data.

In [146]:
# Display Freq Table Percentage for App Store

app_store_genres = display_table(app_store_final, -5)
print(app_store_genres)

Games : 58.16
Entertainment : 7.88
Photo & Video : 4.97
Education : 3.66
Social Networking : 3.29
Shopping : 2.61
Utilities : 2.51
Sports : 2.14
Music : 2.05
Health & Fitness : 2.02
Productivity : 1.74
Lifestyle : 1.58
News : 1.33
Travel : 1.24
Finance : 1.12
Weather : 0.87
Food & Drink : 0.81
Reference : 0.56
Business : 0.53
Book : 0.43
Navigation : 0.19
Medical : 0.19
Catalogs : 0.12
None


Our analysis tells us that among free apps aimed at an English speaking audience, more than half (58.16%) are games followed by Entertainment at about 8%, Photo & Video at about 5%, and Education and Social Networking aroun 3%. 

We can conclude that the App Store apps are dominated by gaming apps, while apps with a practical purpose (Shopping, Education, Productivity) are more rare. 

Although gaming apps are the most frequently developed, this does not mean that they are the ones with the greatest number of users.

## Google Play Genres

In the Google Play data we identified two columns that could be of use: 'Category' and 'Genres'. Let us analyze both of these. 

In [149]:
# Analyze 'Category' column of Google Play data

gplay_cats = display_table(gplay_final, 1)
print(gplay_cats)

FAMILY : 18.91
GAME : 9.72
TOOLS : 8.46
BUSINESS : 4.59
LIFESTYLE : 3.9
PRODUCTIVITY : 3.89
FINANCE : 3.7
MEDICAL : 3.53
SPORTS : 3.4
PERSONALIZATION : 3.32
COMMUNICATION : 3.24
HEALTH_AND_FITNESS : 3.08
PHOTOGRAPHY : 2.94
NEWS_AND_MAGAZINES : 2.8
SOCIAL : 2.66
TRAVEL_AND_LOCAL : 2.34
SHOPPING : 2.25
BOOKS_AND_REFERENCE : 2.14
DATING : 1.86
VIDEO_PLAYERS : 1.79
MAPS_AND_NAVIGATION : 1.4
FOOD_AND_DRINK : 1.24
EDUCATION : 1.16
ENTERTAINMENT : 0.96
LIBRARIES_AND_DEMO : 0.94
AUTO_AND_VEHICLES : 0.93
HOUSE_AND_HOME : 0.82
WEATHER : 0.8
EVENTS : 0.71
PARENTING : 0.65
ART_AND_DESIGN : 0.64
COMICS : 0.62
BEAUTY : 0.6
None


The Google Play data tells a completely different story. Google Play apps or dominated by apps with practical applications (Family: 19%, Tools: 8.5%, Business: 4.6%) while gaming apps are at about 10%. 

We can now analyze the 'Genres' column. 

In [152]:
# Analyze 'Genres' column of Google Play data

gplay_genres = display_table(gplay_final, -4)
print(gplay_genres)

Tools : 8.45
Entertainment : 6.07
Education : 5.35
Business : 4.59
Productivity : 3.89
Lifestyle : 3.89
Finance : 3.7
Medical : 3.53
Sports : 3.46
Personalization : 3.32
Communication : 3.24
Action : 3.1
Health & Fitness : 3.08
Photography : 2.94
News & Magazines : 2.8
Social : 2.66
Travel & Local : 2.32
Shopping : 2.25
Books & Reference : 2.14
Simulation : 2.04
Dating : 1.86
Arcade : 1.85
Video Players & Editors : 1.77
Casual : 1.76
Maps & Navigation : 1.4
Food & Drink : 1.24
Puzzle : 1.13
Racing : 0.99
Role Playing : 0.94
Libraries & Demo : 0.94
Auto & Vehicles : 0.93
Strategy : 0.91
House & Home : 0.82
Weather : 0.8
Events : 0.71
Adventure : 0.68
Comics : 0.61
Beauty : 0.6
Art & Design : 0.6
Parenting : 0.5
Card : 0.45
Casino : 0.43
Trivia : 0.42
Educational;Education : 0.39
Board : 0.38
Educational : 0.37
Education;Education : 0.34
Word : 0.26
Casual;Pretend Play : 0.24
Music : 0.2
Racing;Action & Adventure : 0.17
Puzzle;Brain Games : 0.17
Entertainment;Music & Video : 0.17
Casual;

We can see that the 'Genres' column contains much more specific data. At this point we are looking to understand a bigger picture perspective to developing apps. For this reason, we will stick with the analysis of the 'Category' column. 

Now we can tell that the App Store is dominated by games made for fun, while
Google Play is dominated by apps with practical use, but does show some balance with for fun games. We will now move on to analyzing the apps with the most users.

# Dataset Analysis - Apps With Most Users

Given the conclusions reached in the analysis on most popular genres/categories of apps, we can now find out which genres/categories have the most users. 

One way to achieve this is by calculating the average number of installs for each app genre/category. For the Google Play data we can find this information in the 'Installs' column. The App Store data is missing this information, but we will use the 'rating_count_tot' column to conduct our analysis. 

## App Store - Number of Users per Genre

To analyze the App Store data, we will: 

- Take the genre of each app
- Sum the number of user ratings for the apps of that genre 
- Divide by the number of apps belonging to that genre

To achieve this task we will uses a for loop inside another for loop, a nested loop.

In [183]:
# App Store Genre Frequency

app_store_genres = freq_table(app_store_final, -5)

for genre in app_store_genres:
    total = 0  #store sum of number of ratings
    len_genre = 0  #store number of apps specific to each genre
    
    for app in app_store_final:
        genre_app = app[-5]
        
        if genre_app == genre:
            num_ratings = float(app[5])  #'rating_count_tot' column
            total += num_ratings
            len_genre += 1
            
    avg_num_ratings = total / len_genre
    print(genre, ':', round(avg_num_ratings, 2)) 

Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22788.67
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


We can see that the top three genres by average number of user ratings are:

- Navigation: 86,090
- Reference: 74,942
- Social Networking: 71,548

Let's explore these categories further. 

In [172]:
# Analyze Navigation apps

for app in app_store_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Here we see that the number of user ratings for Navigation apps are highly influenced by Waze and Google Maps accounting for about 500,000 reviews. 

In [174]:
# Analyze Reference apps

for app in app_store_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Reference apps are highly influenced by the Bible and Dictionary apps, about 1.2 million user ratings.

Let's move onto analyzing the Social apps category.

In [177]:
# Analyze Social apps

for app in app_store_final:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

Social Netwokring apps are higly influenced by Facebook and Pinterest: 4 million user ratings.

By looking through the apps of the top three genre categories of the App Store, I see the best opportunity for developing an app with low cost and high potential in the Reference category. 

Developing a new Navigation or Social Networking app takes time, trials, and a lot of resources. But if we look at some of the apps that are more popular in the Reference category, we apps that are guide books for gaming apps. Developing a new guide book for a popular game can be a good opporutnity for the company. 

## Google Play - Number of Users per Genre

We will do the same type of analysis for the Google Play data by analyzing the 'Installs' column. 

In [181]:
display_table(gplay_final, 5)

1,000,000+ : 15.73
100,000+ : 11.55
10,000,000+ : 10.55
10,000+ : 10.2
1,000+ : 8.39
100+ : 6.92
5,000,000+ : 6.83
500,000+ : 5.56
50,000+ : 4.77
5,000+ : 4.51
10+ : 3.54
500+ : 3.25
50,000,000+ : 2.3
100,000,000+ : 2.13
50+ : 1.92
5+ : 0.79
1+ : 0.51
500,000,000+ : 0.27
1,000,000,000+ : 0.23
0+ : 0.05
0 : 0.01


As we can see the number of installs isn't very specific. We can't tell if an app with 100,000+ has 100,000 or 200,000 installs. For our purposes we will consider that 100,000+ is equal to 100,000 installs and an app with 1,000,000+ is equal to 1,000,000 installs. 

In order to perform calcualtions on this, we need to remove the commas and plus character and convert the string into a float first. 

In [190]:
# Google Play Category Frequency

gplay_cats = freq_table(gplay_final, 1)

for cat in gplay_cats:
    total = 0  
    len_cat = 0  
    
    for app in gplay_final:
        app_cat = app[1]
        
        if app_cat == cat:
            num_installs = app[5]
            
            num_installs = num_installs.replace(',','')
            num_installs = num_installs.replace('+', '')
            
            total += float(num_installs)
            len_cat += 1
            
    avg_num_installs = total / len_cat
    print(cat, ':', round(avg_num_installs, 2))

ART_AND_DESIGN : 1986335.09
AUTO_AND_VEHICLES : 647317.82
BEAUTY : 513151.89
BOOKS_AND_REFERENCE : 8767811.89
BUSINESS : 1712290.15
COMICS : 817657.27
COMMUNICATION : 38456119.17
DATING : 854028.83
EDUCATION : 1833495.15
ENTERTAINMENT : 11640705.88
EVENTS : 253542.22
FINANCE : 1387692.48
FOOD_AND_DRINK : 1924897.74
HEALTH_AND_FITNESS : 4188821.99
HOUSE_AND_HOME : 1331540.56
LIBRARIES_AND_DEMO : 638503.73
LIFESTYLE : 1437816.27
GAME : 15588015.6
FAMILY : 3695641.82
MEDICAL : 120550.62
SOCIAL : 23253652.13
SHOPPING : 7036877.31
PHOTOGRAPHY : 17840110.4
SPORTS : 3638640.14
TRAVEL_AND_LOCAL : 13984077.71
TOOLS : 10801391.3
PERSONALIZATION : 5201482.61
PRODUCTIVITY : 16787331.34
PARENTING : 542603.62
WEATHER : 5074486.2
VIDEO_PLAYERS : 24727872.45
NEWS_AND_MAGAZINES : 9549178.47
MAPS_AND_NAVIGATION : 4056941.77


The top three categories for apps in Google Play in terms of installs are:

- Communication: 38,456,119
- Video Players: 24,727,872
- Social: 23,253,652

If we explore these categories further, we will find the same pattern present in the App Store. A few dominate the market and will be hard to compete against. 

In [193]:
# Analyze Communication apps

for app in gplay_final:
    if app[1] == 'COMMUNICATION':
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

Apps like Facebook Messenger, WhatsApp, and Skype have over a billion installs with a few others having over 100 and 500 million installs.  

In [195]:
# Analyze Video Players apps

for app in gplay_final:
    if app[1] == 'VIDEO_PLAYERS':
        print(app[0], ':', app[5])

YouTube : 1,000,000,000+
All Video Downloader 2018 : 1,000,000+
Video Downloader : 10,000,000+
HD Video Player : 1,000,000+
Iqiyi (for tablet) : 1,000,000+
Video Player All Format : 10,000,000+
Motorola Gallery : 100,000,000+
Free TV series : 100,000+
Video Player All Format for Android : 500,000+
VLC for Android : 100,000,000+
Code : 10,000,000+
Vote for : 50,000,000+
XX HD Video downloader-Free Video Downloader : 1,000,000+
OBJECTIVE : 1,000,000+
Music - Mp3 Player : 10,000,000+
HD Movie Video Player : 1,000,000+
YouCut - Video Editor & Video Maker, No Watermark : 5,000,000+
Video Editor,Crop Video,Movie Video,Music,Effects : 1,000,000+
YouTube Studio : 10,000,000+
video player for android : 10,000,000+
Vigo Video : 50,000,000+
Google Play Movies & TV : 1,000,000,000+
HTC Service － DLNA : 10,000,000+
VPlayer : 1,000,000+
MiniMovie - Free Video and Slideshow Editor : 50,000,000+
Samsung Video Library : 50,000,000+
OnePlus Gallery : 1,000,000+
LIKE – Magic Video Maker & Community : 50,

Apps like YouTube and Google Play Movies & TV have over a billion downloads with some having 100 million and 500 million installs.

In [196]:
# Analyze Social apps

for app in gplay_final:
    if app[1] == 'SOCIAL':
        print(app[0], ':', app[5])

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Social network all in one 2018 : 100,000+
Pinterest : 100,000,000+
TextNow - free text + calls : 10,000,000+
Google+ : 1,000,000,000+
The Messenger App : 1,000,000+
Messenger Pro : 1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus : 1,000,000+
Telegram X : 5,000,000+
The Video Messenger App : 100,000+
Jodel - The Hyperlocal App : 1,000,000+
Hide Something - Photo, Video : 5,000,000+
Love Sticker : 1,000,000+
Web Browser & Fast Explorer : 5,000,000+
LiveMe - Video chat, new friends, and make money : 10,000,000+
VidStatus app - Status Videos & Status Downloader : 5,000,000+
Love Images : 1,000,000+
Web Browser ( Fast & Secure Web Explorer) : 500,000+
SPARK - Live random video chat & meet new people : 5,000,000+
Golden telegram : 50,000+
Facebook Local : 1,000,000+
Meet – Talk to Strangers Using Random Video Chat : 5,000,000+
MobilePatrol Public Safety App : 1,000,000+
💘 WhatsLov: Smileys of love, sti

In this arena we would have to compete against Facebook and Instagram which have a strong hold on the market. 

Seeing that these top 3 categories seem to saturated and a few players have a large market share, let's analyze the Book & Reference category that we found to be a viable market in the App Store. 

This market has about 9 million installs and doesn't seem too saturated

In [198]:
# Analyze Book & Reference apps

for app in gplay_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

There are a great variety of apps in this category as we can see. We do see the same popularity of the Bible and Dictionary apps here as well as a niche segment for installs of guides to popular games. 

# Project Conclusion

In this project we analyzed app data from both Google Play and the App Store with the goal of identifying app profiles of the type of apps that would attract the most users. 

We concluded that developing a guide book for a popular gaming app could be profitable in both the Google Play and App Store markets. Other categories apps showed higher numbers of users, but given our strategy of minimizing risk, costs, and time to develop, the Book & Refernce sector shows the most promise. Specifically when considering that in other categories/genres we would have to compete against giants such as Facebook, Instagram, Spotify, and YouTube. 

We can look at which games are popular in both markets and develop a guide book for them. 

In [205]:
# Analyze Game Apps Google Play

for app in gplay_final:
    if app[1] == 'GAME'   and (app[5] == '100,000,000+'
                                            or app[5] == '50,000,000+'
                                            or app[5] == '10,000,000+'):
        print(app[0], ':', app[5])

Solitaire : 10,000,000+
Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Bubble Witch 3 Saga : 50,000,000+
Race the Traffic Moto : 10,000,000+
Marble - Temple Quest : 10,000,000+
Shooting King : 10,000,000+
Geometry Dash World : 10,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Block Craft 3D: Building Simulator Games For Free : 50,000,000+
Love Balls : 50,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Paint Hit : 10,000,000+
Snake VS Block : 50,000,000+
Rolly Vortex : 10,000,000+
Stack Jump : 10,000,000+
Extreme Car Driving Simulator : 100,000,000+
Color Road : 10,000,000+
Draw In : 10,000,000+
Trivia Crack : 100,000,000+
Baseball Boy! : 10,000,000+
Hello Stars : 10,000,000+
Tank Stars : 10,000,000+
Hole.io : 10,000,000+
Flip the Gun - Simulator Game : 10,000,000+
Word Link : 10,000,000+
Last Day on Earth: Survival : 10,000,000+
Partymasters - Fun Idle Game : 10,000,000+
Harry Potter: Hogwarts Mystery : 10,000,000+
Cooking Madness - A Chef's Restaurant Games :

Let's look at the App Store next.

In [207]:
# Analyze Games apps

for app in app_store_final:
    if app[-5] == 'Games':
        print(app[1], ':', app[5])

Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO : 257627
CSR Racing 2 : 257100
Star Wars™: 

The Walls : 1662
Dulp : 1660
Kerflux : 1648
Epic Battle Simulator : 1647
Dead Zone: Zombie Crisis : 1632
Drifty Chase : 1631
Splash : 1631
Coco Pony - My Dream Pet : 1627
Slide the Shakes : 1623
Coach Bus Simulator : 1621
Hockey Stars : 1619
Pixduel™ : 1602
Tap Tap Trillionaire : 1595
Super Stickman Golf 3 : 1593
Lastronaut : 1576
Cops N Robbers (Jail Break 2) - Survival Mini Game : 1559
Microtrip : 1551
Sir Match-a-Lot – Match 3 Puzzle Game : 1546
Disney Jigsaw Puzzles! : 1546
Snow Roll : 1533
Ice Age: Arctic Blast : 1523
Into the Dim : 1521
Forest Rescue 2: Friends United Match 3 Puzzle : 1505
Knights Fight: Medieval Arena : 1501
Pathfinder Adventures : 1498
Cafe Story - Play Cooking & Farming Game : 1498
Dunkers : 1495
Hidden Pictures by Highlights Magazine : 1478
Vintage Slots Las Vegas - Old Slot Machine Games! : 1474
Slots: Fast Fortune Slot Machines & Fun Slot Games : 1474
Outfolded : 1468
LongStory: Choose your own dating game : 1467
Tropical Wars - Pirate Battles : 1465
360 De

Crazy Eye Clinic - Doctor X Adventures : 101
Don't Be Squared : 100
Street Fighter Puzzle Spirits : 100
Panic Drop : 99
Extreme Forklifting 2 : 99
Princess Libby: Frozen Party : 99
Maim Street : 99
Rogue Ninja : 99
Level With Me : 98
Princess Libby's Perfect Beach Day : 98
Crazy Circle : 98
A.P.E.X : 96
Christmas Swap 3 -Match toy & candy to countdown : 96
Pigu : 94
Pop Girls - High School Band : 94
Fairy Land Rescue - Save the Magic Village : 94
KOMRAD : 94
Water Bottle Flip AK 2016 : 93
Danger Dodgers : 93
Hugo Troll Race 2. : 91
Krosmaster Arena : 91
Boom Boom Soccer : 91
Baby Full House - Care, Play and Have Fun : 91
Sausage Legend - Fighting Game : 90
Colorpede : 90
Glitch Fixers - The Powerpuff Girls : 89
PopSwish Basketball by Andrew Wiggins : 89
Giant Turnip Game: A Voyage Of Vegetable Extraction! : 89
Thomas & Friends: Race On! : 89
Jurassic Virtual Reality (VR) : 88
Roof Jumping 3 Stunt Driver Parking Simulator an Extreme Real Car Racing Game : 88
Ducky Fuzz - Chain Reaction 

In both markets we find that Clash of Clans is one of the more popular games. So developing a comprehensive guide book as to how to be succesful in the game should prove profitable for the company.